### PR TITLE
Refactor Payment to OnlinePayment

### PR DESCRIPTION
**Pull request by MW-Forster**

### CODE REVIEW

File: src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpOnlinePaymentsServiceTests.cs [hunk 1]
```diff
@@ -34,15 +34,15 @@ public void Initialize()
             var config = new Service
             {
                 Url = "https://example.com",
-                EndPointName = "payments",
+                EndPointName = "onlinepayments",
```
Comment (davidjayakumarEviden) [inline src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpOnlinePaymentsServiceTests.cs:37]: Should be online-payment

File: src/EPR.Payment.Facade/wwwroot/index.html [hunk 2]
```diff
@@ -227,7 +227,7 @@
             console.log('Request Data:', requestData);
 
             try {
-                const response = await fetch('/api/v1/payments', {
+                const response = await fetch('/api/v1/onlinepayments', {
```
Comment (davidjayakumarEviden) [inline src/EPR.Payment.Facade/wwwroot/index.html:230]: Same, should be online-payments

File: src/EPR.Payment.Facade/Controllers/Payments/OnlinePaymentsController.cs [hunk 3]
```diff
@@ -14,19 +14,19 @@ namespace EPR.Payment.Facade.Controllers.Payments
 {
     [ApiVersion(1)]
     [ApiController]
-    [Route("api/v{version:apiVersion}/payments")]
+    [Route("api/v{version:apiVersion}/onlinepayments")]
```
Comment (davidjayakumarEviden) [inline src/EPR.Payment.Facade/Controllers/Payments/OnlinePaymentsController.cs:17]: Same, should be online-payments

**Review by davidjayakumarEviden — APPROVED** [review]
Looks good

File: src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpOnlinePaymentsServiceTests.cs (hunk 1 above)
Comment (zozanKirkutay) [reply src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpOnlinePaymentsServiceTests.cs]: it is EndPointName = "online-payments", now

File: src/EPR.Payment.Facade/wwwroot/index.html (hunk 2 above)
Comment (zozanKirkutay) [reply src/EPR.Payment.Facade/wwwroot/index.html]: in onlinePayment.js it is =>  const response = await fetch('/api/v1/online-payments',

File: src/EPR.Payment.Facade/Controllers/Payments/OnlinePaymentsController.cs (hunk 3 above)
Comment (zozanKirkutay) [reply src/EPR.Payment.Facade/Controllers/Payments/OnlinePaymentsController.cs]: it is  [Route("api/v{version:apiVersion}/online-payments")] now

File: src/EPR.Payment.Facade/Services/Payments/OnlinePaymentsService.cs [hunk 4]
```diff
@@ -116,62 +116,63 @@ private async Task<PaymentStatusResponseDto> GetPaymentStatusResponseAsync(strin
...
-                _logger.LogError(ex, LogMessages.UnexpectedErrorUpdatingPayment);
-                throw new ServiceException(ExceptionMessages.UnexpectedErrorUpdatingPayment, ex);
+                _logger.LogError(ex, LogMessages.UnexpectedErrorUpdatingOnlinePayment);
+                throw new ServiceException(ExceptionMessages.UnexpectedErrorUpdatingOnlinePayment, ex);
             }
         }
 
 
-        private static CompletePaymentResponseDto CreateCompletePaymentResponse(PaymentDetailsDto paymentDetails, PaymentStatusResponseDto paymentStatusResponse, PaymentStatus status)
+        private CompleteOnlinePaymentResponseDto CreateCompleteOnlinePaymentResponse(OnlinePaymentDetailsDto onlinePaymentDetails, PaymentStatusResponseDto paymentStatusResponse, PaymentStatus status)
```
Comment (manishagarwalEviden) [inline src/EPR.Payment.Facade/Services/Payments/OnlinePaymentsService.cs:151]: sonarqube: Member 'CreateCompleteOnlinePaymentResponse' does not access instance data and can be marked as static

File: src/EPR.Payment.Facade/Services/Payments/OnlinePaymentsService.cs [hunk 5]
```diff
@@ -193,56 +194,56 @@ private async Task<GovPayResponseDto> InitiateGovPayPaymentAsync(GovPayRequestDt
...
             {
-                _logger.LogError(ex, LogMessages.UnexpectedErrorUpdatingPayment);
-                throw new ServiceException(ExceptionMessages.UnexpectedErrorUpdatingPayment, ex);
+                _logger.LogError(ex, LogMessages.UnexpectedErrorUpdatingOnlinePayment);
+                throw new ServiceException(ExceptionMessages.UnexpectedErrorUpdatingOnlinePayment, ex);
             }
         }
 
-        private static PaymentResponseDto CreatePaymentResponse(GovPayResponseDto govPayResponse)
+        private OnlinePaymentResponseDto CreateOnlinePaymentResponse(GovPayResponseDto govPayResponse)
```
Comment (manishagarwalEviden) [inline src/EPR.Payment.Facade/Services/Payments/OnlinePaymentsService.cs:220]: sonarqube: Member 'CreateOnlinePaymentResponse' does not access instance data and can be marked as static

**Review by manishagarwalEviden — CHANGES_REQUESTED** [review]

**Review by paulshimmen — CHANGES_REQUESTED** [review]

File: src/EPR.Payment.Facade.Common/Constants/ExceptionMessages.cs [hunk 6]
```diff
@@ -10,15 +10,15 @@ public static class ExceptionMessages
...
-        public const string PaymentServiceHttpClientNameMissing = "PaymentService HttpClientName configuration is missing";
-        public const string ErrorInsertingPayment = "Error occurred while inserting payment status.";
-        public const string ErrorUpdatingPayment = "Error occurred while updating payment status.";
-        public const string UnexpectedErrorInsertingPayment = "An unexpected error occurred while inserting the payment.";
-        public const string UnexpectedErrorUpdatingPayment = "An unexpected error occurred while updating the payment status.";
-        public const string ErrorRetrievingPaymentDetails = "Error occurred while retrieving payment details.";
-        public const string ErrorGettingPaymentDetails = "Error occurred while getting payment details.";
+        public const string OnlinePaymentServiceBaseUrlMissing = "PaymentService BaseUrl configuration is missing";
+        public const string OnlinePaymentServiceEndPointNameMissing = "PaymentService EndPointName configuration is missing";
+        public const string OnlinePaymentServiceHttpClientNameMissing = "PaymentService HttpClientName configuration is missing";
```
[the remark this answers is not in the document]
Comment (zozanKirkutay) [reply src/EPR.Payment.Facade.Common/Constants/ExceptionMessages.cs]: done

File: src/EPR.Payment.Facade/Services/Payments/OnlinePaymentsService.cs (hunk 4 above)
Comment (zozanKirkutay) [reply src/EPR.Payment.Facade/Services/Payments/OnlinePaymentsService.cs]: done

File: src/EPR.Payment.Facade/Services/Payments/OnlinePaymentsService.cs (hunk 5 above)
Comment (zozanKirkutay) [reply src/EPR.Payment.Facade/Services/Payments/OnlinePaymentsService.cs]: done

**Review by paulshimmen — APPROVED** [review]

**Pull request merged**